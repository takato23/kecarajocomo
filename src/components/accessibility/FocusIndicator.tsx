'use client';

import { useEffect, useState } from 'react';

export function FocusIndicator() {
  const [isKeyboardUser, setIsKeyboardUser] = useState(false);

  useEffect(() => {
    // Detect keyboard navigation
    const handleMouseDown = () => setIsKeyboardUser(false);
    const handleKeyDown = (e: KeyboardEvent) => {
      if (e.key === 'Tab') {
        setIsKeyboardUser(true);
      }
    };

    document.addEventListener('mousedown', handleMouseDown);
    document.addEventListener('keydown', handleKeyDown);

    // Add focus-visible class to body when keyboard navigation is detected
    if (isKeyboardUser) {
      document.body.classList.add('keyboard-navigation');
    } else {
      document.body.classList.remove('keyboard-navigation');
    }

    return () => {
      document.removeEventListener('mousedown', handleMouseDown);
      document.removeEventListener('keydown', handleKeyDown);
    };
  }, [isKeyboardUser]);

  return null;
}