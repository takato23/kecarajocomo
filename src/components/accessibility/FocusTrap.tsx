'use client';

import { useRef, useEffect } from 'react';
import FocusTrapReact from 'focus-trap-react';

interface FocusTrapProps {
  children: React.ReactNode;
  active?: boolean;
  paused?: boolean;
  onDeactivate?: () => void;
  initialFocus?: string | false;
  fallbackFocus?: string;
  escapeDeactivates?: boolean;
  clickOutsideDeactivates?: boolean;
}

export function FocusTrap({
  children,
  active = true,
  paused = false,
  onDeactivate,
  initialFocus,
  fallbackFocus,
  escapeDeactivates = true,
  clickOutsideDeactivates = true,
}: FocusTrapProps) {
  const trapRef = useRef<any>(null);

  useEffect(() => {
    if (!active && trapRef.current && typeof trapRef.current.deactivate === 'function') {
      trapRef.current.deactivate();
    }
  }, [active]);

  return (
    <FocusTrapReact
      ref={trapRef}
      active={active}
      paused={paused}
      focusTrapOptions={{
        onDeactivate,
        initialFocus,
        fallbackFocus,
        escapeDeactivates,
        clickOutsideDeactivates,
        returnFocusOnDeactivate: true,
        allowOutsideClick: true,
      }}
    >
      {children}
    </FocusTrapReact>
  );
}