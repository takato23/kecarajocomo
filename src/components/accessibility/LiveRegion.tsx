'use client';

import { useEffect, useRef } from 'react';

interface LiveRegionProps {
  message: string;
  politeness?: 'polite' | 'assertive';
  clearOnUnmount?: boolean;
}

export function LiveRegion({ 
  message, 
  politeness = 'polite',
  clearOnUnmount = true 
}: LiveRegionProps) {
  const regionRef = useRef<HTMLDivElement>(null);
  const previousMessage = useRef<string>('');

  useEffect(() => {
    // Only announce if message has changed
    if (message && message !== previousMessage.current && regionRef.current) {
      // Clear and re-set the content to ensure screen readers announce it
      regionRef.current.textContent = '';
      setTimeout(() => {
        if (regionRef.current) {
          regionRef.current.textContent = message;
        }
      }, 100);
      previousMessage.current = message;
    }
  }, [message]);

  useEffect(() => {
    return () => {
      if (clearOnUnmount && regionRef.current) {
        regionRef.current.textContent = '';
      }
    };
  }, [clearOnUnmount]);

  return (
    <div 
      ref={regionRef}
      role="status" 
      aria-live={politeness} 
      aria-atomic="true"
      className="sr-only"
    />
  );
}

// Hook for using live regions
export function useLiveRegion() {
  const regionRef = useRef<HTMLDivElement | null>(null);
  const mounted = useRef(false);

  useEffect(() => {
    if (!mounted.current) {
      // Create a live region element and append it to the body
      const liveRegion = document.createElement('div');
      liveRegion.setAttribute('role', 'status');
      liveRegion.setAttribute('aria-live', 'polite');
      liveRegion.setAttribute('aria-atomic', 'true');
      liveRegion.className = 'sr-only';
      document.body.appendChild(liveRegion);
      regionRef.current = liveRegion;
      mounted.current = true;
    }

    return () => {
      if (regionRef.current && regionRef.current.parentNode) {
        regionRef.current.parentNode.removeChild(regionRef.current);
      }
    };
  }, []);

  const announce = (message: string, politeness: 'polite' | 'assertive' = 'polite') => {
    if (regionRef.current) {
      regionRef.current.setAttribute('aria-live', politeness);
      regionRef.current.textContent = '';
      setTimeout(() => {
        if (regionRef.current) {
          regionRef.current.textContent = message;
        }
      }, 100);
    }
  };

  return { announce };
}