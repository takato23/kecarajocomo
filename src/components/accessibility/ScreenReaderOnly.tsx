import { ReactNode } from 'react';

interface ScreenReaderOnlyProps {
  children: ReactNode;
  as?: keyof JSX.IntrinsicElements;
}

export function ScreenReaderOnly({ 
  children, 
  as: Component = 'span' 
}: ScreenReaderOnlyProps) {
  return (
    <Component className="sr-only">
      {children}
    </Component>
  );
}