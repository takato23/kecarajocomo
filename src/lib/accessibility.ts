/**
 * Accessibility utility functions
 */

// Generate unique IDs for ARIA relationships
export function generateId(prefix: string = 'id'): string {
  return `${prefix}-${Math.random().toString(36).substr(2, 9)}`;
}

// Check if user prefers reduced motion
export function prefersReducedMotion(): boolean {
  if (typeof window === 'undefined') return false;
  const mediaQuery = window.matchMedia('(prefers-reduced-motion: reduce)');
  return mediaQuery.matches;
}

// Check if user prefers high contrast
export function prefersHighContrast(): boolean {
  if (typeof window === 'undefined') return false;
  const mediaQuery = window.matchMedia('(prefers-contrast: high)');
  return mediaQuery.matches;
}

// Get user's preferred color scheme
export function getPreferredColorScheme(): 'light' | 'dark' | 'no-preference' {
  if (typeof window === 'undefined') return 'no-preference';
  
  if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
    return 'dark';
  }
  if (window.matchMedia('(prefers-color-scheme: light)').matches) {
    return 'light';
  }
  return 'no-preference';
}

// Trap focus within an element
export function trapFocus(element: HTMLElement) {
  const focusableElements = element.querySelectorAll(
    'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex="0"], [contenteditable]'
  );
  
  const firstFocusableElement = focusableElements[0] as HTMLElement;
  const lastFocusableElement = focusableElements[focusableElements.length - 1] as HTMLElement;

  const handleTabKey = (e: KeyboardEvent) => {
    if (e.key !== 'Tab') return;

    if (e.shiftKey) {
      if (document.activeElement === firstFocusableElement) {
        lastFocusableElement.focus();
        e.preventDefault();
      }
    } else {
      if (document.activeElement === lastFocusableElement) {
        firstFocusableElement.focus();
        e.preventDefault();
      }
    }
  };

  element.addEventListener('keydown', handleTabKey);
  firstFocusableElement?.focus();

  return () => {
    element.removeEventListener('keydown', handleTabKey);
  };
}

// Announce message to screen readers
export function announce(message: string, priority: 'polite' | 'assertive' = 'polite') {
  const announcement = document.createElement('div');
  announcement.setAttribute('aria-live', priority);
  announcement.setAttribute('aria-atomic', 'true');
  announcement.setAttribute('class', 'sr-only');
  announcement.textContent = message;

  document.body.appendChild(announcement);

  setTimeout(() => {
    document.body.removeChild(announcement);
  }, 1000);
}

// Alias for announce function (for backward compatibility)
export const announceToScreenReader = announce;

// Format time for screen readers
export function formatTimeForScreenReader(minutes: number): string {
  const hours = Math.floor(minutes / 60);
  const mins = minutes % 60;
  
  if (hours > 0) {
    return `${hours} hour${hours > 1 ? 's' : ''} and ${mins} minute${mins !== 1 ? 's' : ''}`;
  }
  return `${mins} minute${mins !== 1 ? 's' : ''}`;
}

// Create ARIA description from multiple sources
export function createAriaDescription(...descriptions: (string | undefined)[]): string | undefined {
  const filtered = descriptions.filter(Boolean);
  return filtered.length > 0 ? filtered.join('. ') : undefined;
}

// Check if element is visible
export function isElementVisible(element: HTMLElement): boolean {
  const rect = element.getBoundingClientRect();
  const style = window.getComputedStyle(element);
  
  return (
    rect.width > 0 &&
    rect.height > 0 &&
    style.display !== 'none' &&
    style.visibility !== 'hidden' &&
    style.opacity !== '0'
  );
}

// Get focusable elements within a container
export function getFocusableElements(container: HTMLElement): HTMLElement[] {
  const elements = container.querySelectorAll(
    'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex]:not([tabindex="-1"]), [contenteditable]'
  );
  
  return Array.from(elements).filter(el => isElementVisible(el as HTMLElement)) as HTMLElement[];
}

// Get first focusable element
export function getFirstFocusableElement(container: HTMLElement): HTMLElement | null {
  const elements = getFocusableElements(container);
  return elements.length > 0 ? elements[0] : null;
}

// Get last focusable element
export function getLastFocusableElement(container: HTMLElement): HTMLElement | null {
  const elements = getFocusableElements(container);
  return elements.length > 0 ? elements[elements.length - 1] : null;
}

// Alias for getFocusableElements (for backward compatibility)
export const getAllFocusableElements = getFocusableElements;