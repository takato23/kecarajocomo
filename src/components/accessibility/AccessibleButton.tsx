'use client';

import { forwardRef, ButtonHTMLAttributes } from 'react';

import { cn } from '@/lib/utils';

import { ScreenReaderOnly } from './ScreenReaderOnly';

interface AccessibleButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: 'primary' | 'secondary' | 'outline' | 'ghost' | 'danger';
  size?: 'sm' | 'md' | 'lg';
  loading?: boolean;
  loadingText?: string;
  icon?: React.ReactNode;
  iconPosition?: 'left' | 'right';
  fullWidth?: boolean;
  ariaLabel?: string;
  ariaPressed?: boolean;
  ariaExpanded?: boolean;
  ariaControls?: string;
  srOnlyText?: string;
}

const buttonVariants = {
  primary: 'bg-primary text-primary-foreground hover:bg-primary/90 focus:ring-primary',
  secondary: 'bg-secondary text-secondary-foreground hover:bg-secondary/80 focus:ring-secondary',
  outline: 'border border-input bg-background hover:bg-accent hover:text-accent-foreground focus:ring-accent',
  ghost: 'hover:bg-accent hover:text-accent-foreground focus:ring-accent',
  danger: 'bg-destructive text-destructive-foreground hover:bg-destructive/90 focus:ring-destructive',
};

const buttonSizes = {
  sm: 'h-8 px-3 text-sm',
  md: 'h-10 px-4',
  lg: 'h-12 px-6 text-lg',
};

export const AccessibleButton = forwardRef<HTMLButtonElement, AccessibleButtonProps>(
  (
    {
      className,
      variant = 'primary',
      size = 'md',
      loading = false,
      loadingText = 'Loading...',
      icon,
      iconPosition = 'left',
      fullWidth = false,
      ariaLabel,
      ariaPressed,
      ariaExpanded,
      ariaControls,
      srOnlyText,
      disabled,
      children,
      ...props
    },
    ref
  ) => {
    const isDisabled = disabled || loading;

    return (
      <button
        ref={ref}
        className={cn(
          'inline-flex items-center justify-center rounded-md font-medium transition-colors',
          'focus:outline-none focus:ring-2 focus:ring-offset-2',
          'disabled:pointer-events-none disabled:opacity-50',
          buttonVariants[variant],
          buttonSizes[size],
          fullWidth && 'w-full',
          className
        )}
        disabled={isDisabled}
        aria-label={ariaLabel}
        aria-pressed={ariaPressed}
        aria-expanded={ariaExpanded}
        aria-controls={ariaControls}
        aria-busy={loading}
        {...props}
      >
        {loading ? (
          <>
            <span className="mr-2 h-4 w-4 animate-spin rounded-full border-2 border-current border-t-transparent" />
            <span aria-live="polite">{loadingText}</span>
          </>
        ) : (
          <>
            {icon && iconPosition === 'left' && (
              <span className="mr-2" aria-hidden="true">
                {icon}
              </span>
            )}
            {children}
            {icon && iconPosition === 'right' && (
              <span className="ml-2" aria-hidden="true">
                {icon}
              </span>
            )}
          </>
        )}
        {srOnlyText && <ScreenReaderOnly>{srOnlyText}</ScreenReaderOnly>}
      </button>
    );
  }
);

AccessibleButton.displayName = 'AccessibleButton';