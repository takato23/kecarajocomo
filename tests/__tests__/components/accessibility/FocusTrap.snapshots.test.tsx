import React from 'react';
import { render } from '@testing-library/react';
import { FocusTrap } from '@/components/accessibility/FocusTrap';

// Mock the focus-trap-react library
jest.mock('focus-trap-react', () => {
  return function MockFocusTrap({ children, active, paused, focusTrapOptions }: any) {
    return (
      <div 
        data-testid="focus-trap"
        data-active={active}
        data-paused={paused}
        data-escape-deactivates={focusTrapOptions?.escapeDeactivates}
        data-click-outside-deactivates={focusTrapOptions?.clickOutsideDeactivates}
        data-return-focus-on-deactivate={focusTrapOptions?.returnFocusOnDeactivate}
        data-allow-outside-click={focusTrapOptions?.allowOutsideClick}
        data-initial-focus={focusTrapOptions?.initialFocus}
        data-fallback-focus={focusTrapOptions?.fallbackFocus}
      >
        {children}
      </div>
    );
  };
});

describe('FocusTrap Snapshots', () => {
  const MockContent = ({ title }: { title: string }) => (
    <div className="mock-content">
      <h2>{title}</h2>
      <button>Button 1</button>
      <input type="text" placeholder="Text input" />
      <button>Button 2</button>
    </div>
  );

  describe('Basic FocusTrap', () => {
    it('should render basic focus trap with default props', () => {
      const { container } = render(
        <FocusTrap>
          <MockContent title="Basic Focus Trap" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render active focus trap', () => {
      const { container } = render(
        <FocusTrap active={true}>
          <MockContent title="Active Focus Trap" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render inactive focus trap', () => {
      const { container } = render(
        <FocusTrap active={false}>
          <MockContent title="Inactive Focus Trap" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('FocusTrap States', () => {
    it('should render paused focus trap', () => {
      const { container } = render(
        <FocusTrap paused={true}>
          <MockContent title="Paused Focus Trap" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render active but paused focus trap', () => {
      const { container } = render(
        <FocusTrap active={true} paused={true}>
          <MockContent title="Active but Paused Focus Trap" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render inactive and paused focus trap', () => {
      const { container } = render(
        <FocusTrap active={false} paused={true}>
          <MockContent title="Inactive and Paused Focus Trap" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('FocusTrap with Callbacks', () => {
    it('should render focus trap with onDeactivate callback', () => {
      const { container } = render(
        <FocusTrap onDeactivate={() => {}}>
          <MockContent title="Focus Trap with Callback" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render focus trap with all callbacks', () => {
      const { container } = render(
        <FocusTrap 
          onDeactivate={() => {}}
          active={true}
          paused={false}
        >
          <MockContent title="Focus Trap with All Callbacks" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('FocusTrap with Focus Options', () => {
    it('should render focus trap with initialFocus selector', () => {
      const { container } = render(
        <FocusTrap initialFocus="#initial-focus">
          <MockContent title="Initial Focus Trap" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render focus trap with fallbackFocus selector', () => {
      const { container } = render(
        <FocusTrap fallbackFocus="#fallback-focus">
          <MockContent title="Fallback Focus Trap" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render focus trap with both focus selectors', () => {
      const { container } = render(
        <FocusTrap 
          initialFocus="#initial-focus"
          fallbackFocus="#fallback-focus"
        >
          <MockContent title="Both Focus Selectors" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render focus trap with disabled initialFocus', () => {
      const { container } = render(
        <FocusTrap initialFocus={false}>
          <MockContent title="Disabled Initial Focus" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('FocusTrap with Deactivation Options', () => {
    it('should render focus trap with escape deactivation disabled', () => {
      const { container } = render(
        <FocusTrap escapeDeactivates={false}>
          <MockContent title="No Escape Deactivation" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render focus trap with click outside deactivation disabled', () => {
      const { container } = render(
        <FocusTrap clickOutsideDeactivates={false}>
          <MockContent title="No Click Outside Deactivation" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render focus trap with all deactivation options disabled', () => {
      const { container } = render(
        <FocusTrap 
          escapeDeactivates={false}
          clickOutsideDeactivates={false}
        >
          <MockContent title="All Deactivation Disabled" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('FocusTrap Complex Configurations', () => {
    it('should render focus trap with complete configuration', () => {
      const { container } = render(
        <FocusTrap
          active={true}
          paused={false}
          onDeactivate={() => {}}
          initialFocus="#start-here"
          fallbackFocus="#fallback-here"
          escapeDeactivates={true}
          clickOutsideDeactivates={true}
        >
          <MockContent title="Complete Configuration" />
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render focus trap for modal-like usage', () => {
      const { container } = render(
        <FocusTrap
          active={true}
          paused={false}
          onDeactivate={() => {}}
          initialFocus="#modal-close"
          escapeDeactivates={true}
          clickOutsideDeactivates={true}
        >
          <div className="modal">
            <h2>Modal Title</h2>
            <p>Modal content</p>
            <button id="modal-close">Close</button>
            <button>Save</button>
          </div>
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render focus trap for form-like usage', () => {
      const { container } = render(
        <FocusTrap
          active={true}
          initialFocus="#first-field"
          fallbackFocus="#submit-button"
          escapeDeactivates={false}
          clickOutsideDeactivates={false}
        >
          <form>
            <input id="first-field" type="text" placeholder="First field" />
            <input type="text" placeholder="Second field" />
            <button id="submit-button" type="submit">Submit</button>
          </form>
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('FocusTrap with Different Content', () => {
    it('should render focus trap with simple text content', () => {
      const { container } = render(
        <FocusTrap>
          <p>Simple text content inside focus trap</p>
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render focus trap with complex nested content', () => {
      const { container } = render(
        <FocusTrap>
          <div className="complex-content">
            <header>
              <h1>Complex Content</h1>
              <nav>
                <a href="#section1">Section 1</a>
                <a href="#section2">Section 2</a>
              </nav>
            </header>
            <main>
              <section id="section1">
                <h2>Section 1</h2>
                <button>Action 1</button>
                <input type="text" placeholder="Input 1" />
              </section>
              <section id="section2">
                <h2>Section 2</h2>
                <button>Action 2</button>
                <input type="text" placeholder="Input 2" />
              </section>
            </main>
          </div>
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render focus trap with no focusable elements', () => {
      const { container } = render(
        <FocusTrap>
          <div>
            <p>No focusable elements here</p>
            <span>Just text content</span>
          </div>
        </FocusTrap>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});