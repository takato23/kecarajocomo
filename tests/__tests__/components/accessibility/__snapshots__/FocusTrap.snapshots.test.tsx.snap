// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusTrap Snapshots Basic FocusTrap should render active focus trap 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      Active Focus Trap
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots Basic FocusTrap should render basic focus trap with default props 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      Basic Focus Trap
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots Basic FocusTrap should render inactive focus trap 1`] = `
<div
  data-active="false"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      Inactive Focus Trap
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap Complex Configurations should render focus trap for form-like usage 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="false"
  data-escape-deactivates="false"
  data-fallback-focus="#submit-button"
  data-initial-focus="#first-field"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <form>
    <input
      id="first-field"
      placeholder="First field"
      type="text"
    />
    <input
      placeholder="Second field"
      type="text"
    />
    <button
      id="submit-button"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap Complex Configurations should render focus trap for modal-like usage 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-initial-focus="#modal-close"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="modal"
  >
    <h2>
      Modal Title
    </h2>
    <p>
      Modal content
    </p>
    <button
      id="modal-close"
    >
      Close
    </button>
    <button>
      Save
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap Complex Configurations should render focus trap with complete configuration 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-fallback-focus="#fallback-here"
  data-initial-focus="#start-here"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      Complete Configuration
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap States should render active but paused focus trap 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-paused="true"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      Active but Paused Focus Trap
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap States should render inactive and paused focus trap 1`] = `
<div
  data-active="false"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-paused="true"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      Inactive and Paused Focus Trap
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap States should render paused focus trap 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-paused="true"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      Paused Focus Trap
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap with Callbacks should render focus trap with all callbacks 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      Focus Trap with All Callbacks
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap with Callbacks should render focus trap with onDeactivate callback 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      Focus Trap with Callback
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap with Deactivation Options should render focus trap with all deactivation options disabled 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="false"
  data-escape-deactivates="false"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      All Deactivation Disabled
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap with Deactivation Options should render focus trap with click outside deactivation disabled 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="false"
  data-escape-deactivates="true"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      No Click Outside Deactivation
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap with Deactivation Options should render focus trap with escape deactivation disabled 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="false"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      No Escape Deactivation
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap with Different Content should render focus trap with complex nested content 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="complex-content"
  >
    <header>
      <h1>
        Complex Content
      </h1>
      <nav>
        <a
          href="#section1"
        >
          Section 1
        </a>
        <a
          href="#section2"
        >
          Section 2
        </a>
      </nav>
    </header>
    <main>
      <section
        id="section1"
      >
        <h2>
          Section 1
        </h2>
        <button>
          Action 1
        </button>
        <input
          placeholder="Input 1"
          type="text"
        />
      </section>
      <section
        id="section2"
      >
        <h2>
          Section 2
        </h2>
        <button>
          Action 2
        </button>
        <input
          placeholder="Input 2"
          type="text"
        />
      </section>
    </main>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap with Different Content should render focus trap with no focusable elements 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div>
    <p>
      No focusable elements here
    </p>
    <span>
      Just text content
    </span>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap with Different Content should render focus trap with simple text content 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <p>
    Simple text content inside focus trap
  </p>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap with Focus Options should render focus trap with both focus selectors 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-fallback-focus="#fallback-focus"
  data-initial-focus="#initial-focus"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      Both Focus Selectors
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap with Focus Options should render focus trap with disabled initialFocus 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-initial-focus="false"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      Disabled Initial Focus
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap with Focus Options should render focus trap with fallbackFocus selector 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-fallback-focus="#fallback-focus"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      Fallback Focus Trap
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;

exports[`FocusTrap Snapshots FocusTrap with Focus Options should render focus trap with initialFocus selector 1`] = `
<div
  data-active="true"
  data-allow-outside-click="true"
  data-click-outside-deactivates="true"
  data-escape-deactivates="true"
  data-initial-focus="#initial-focus"
  data-paused="false"
  data-return-focus-on-deactivate="true"
  data-testid="focus-trap"
>
  <div
    class="mock-content"
  >
    <h2>
      Initial Focus Trap
    </h2>
    <button>
      Button 1
    </button>
    <input
      placeholder="Text input"
      type="text"
    />
    <button>
      Button 2
    </button>
  </div>
</div>
`;
